pkg/resources/perf/cpu.go

```diff
@@ -0,0 +1,142 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package perf
+
+import (
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/prometheus/procfs"
+)
+
+// CPUType is type of Etcd resource.
+const CPUType = resource.Type("CPUStats.perf.talos.dev")
+
+// CPUID is a resource ID of singleton instance.
+const CPUID = resource.ID("latest")
+
+// CPU represents the last CPU stats snapshot.
+type CPU struct {
+	md   resource.Metadata
+	spec CPUSpec
+}
+
+// CPUSpec represents the last CPU stats snapshot.
+type CPUSpec struct {
+	CPU             []*CPUStat `yaml:"cpu"`
```

Comment:
DeepCopy is not correct with this spec, as the spec contains pointers and the actual data behind them is not copied

but instead, I would rather drop pointers here and change to `CPUStat` without `*` (here and on the next line). this should lower GC pressure and make copying easy.